examples/fuzzing/rt_n12/fuzz.py

```diff
@@ -0,0 +1,173 @@
+from qiling import *
+from qiling.const import *
+from qiling.extensions import afl
+from typing import Optional
+from qiling.os.const import *
+from qiling.extensions.coverage import utils as cov_utils
+import argparse
+
+payload_location = 0 # location in snapshot to be modified
+
+def my_send(ql:Qiling):
+    """
+    Hook to avoid writing to now no-longer connected socket
+    used to create this snapshot
+    """
+    params = ql.os.resolve_fcall_params({'sockfd': INT,'buf':POINTER, 'len':SIZE_T, 'flags':INT })
+    ql.log.info(buf)
+    return
+
+def fuzz(ql:Qiling, input_file):
+    def place_input_callback(ql: Qiling, input: bytes, persistent_round: int) -> Optional[bool]:
+        """Feed generated stimuli to the fuzzed target.
+
+        This method is called with every fuzzing iteration.
+        """
+        
+        if len(input)>=10000: # fgets() call uses 10000 for the size arg
+            # so only size-1 bytes will actually be read
+            return False
+               
+        ql.mem.write(payload_location, input)
+    
+        return True
+
+    def start_afl(ql: Qiling):
+        """Have Unicorn fork and start instrumentation.
+        """
+        avoids = []
+        avoids.append(0x0040577c) # 501 not implemented, avoid wasting time
+        # on non POST/HEAD/GET/etc requests
+        avoids.append(0x00405b98) # jump back to select() loop?
+        avoids.append(0x004059b4) # same as above?
+        afl.ql_afl_fuzz(ql, input_file=input_file, place_input_callback=place_input_callback,   exits=avoids)
+    ql.restore(snapshot="httpd.bin")
+    global payload_location
+    payload_location = ql.mem.search(b'GET /FUZZME HTTP/1.1')[0]
+    ql.log.info("Location of input data")
+    ql.log.info(hex(payload_location))
+    ql.os.set_api('send', my_send, QL_INTERCEPT.CALL) # avoid crash due to the socket no longer being valid on clent end
+    ql.hook_address(start_afl, 0x00404b20) # kick off from the start of snapshot
```

Comment:
The `0x00404b20` is magic number that keeps showing up here. Better define it as a variable somewhere with a proper comment that says where it comes from.

Reply:
Agreed.